xbmc/pvr/guilib/PVRGUIActionListener.cpp

```diff
@@ -280,22 +281,32 @@ void CPVRGUIActionListener::OnSettingChanged(const std::shared_ptr<const CSettin
     return;
 
   const std::string& settingId = setting->GetId();
+  const auto& settings = CServiceBroker::GetSettingsComponent()->GetSettings();
   if (settingId == CSettings::SETTING_PVRPARENTAL_ENABLED)
   {
-    if (std::static_pointer_cast<const CSettingBool>(setting)->GetValue() && CServiceBroker::GetSettingsComponent()->GetSettings()->GetString(CSettings::SETTING_PVRPARENTAL_PIN).empty())
+    if (std::static_pointer_cast<const CSettingBool>(setting)->GetValue() &&
+        settings->GetString(CSettings::SETTING_PVRPARENTAL_PIN).empty())
     {
       std::string newPassword = "";
       // password set... save it
       if (CGUIDialogNumeric::ShowAndVerifyNewPassword(newPassword))
-        CServiceBroker::GetSettingsComponent()->GetSettings()->SetString(CSettings::SETTING_PVRPARENTAL_PIN, newPassword);
+        settings->SetString(CSettings::SETTING_PVRPARENTAL_PIN, newPassword);
       // password not set... disable parental
       else
         std::static_pointer_cast<CSettingBool>(std::const_pointer_cast<CSetting>(setting))->SetValue(false);
     }
   }
+  else if (settingId == CSettings::SETTING_EPG_PAST_DAYSTODISPLAY)
+  {
+    int maxFutureDays = settings->GetInt(CSettings::SETTING_EPG_FUTURE_DAYSTODISPLAY);
+    CServiceBroker::GetPVRManager().Clients()->SetEPGTimeFrame(
+        std::static_pointer_cast<const CSettingInt>(setting)->GetValue(), maxFutureDays);
+  }
   else if (settingId == CSettings::SETTING_EPG_FUTURE_DAYSTODISPLAY)
   {
-    CServiceBroker::GetPVRManager().Clients()->SetEPGTimeFrame(std::static_pointer_cast<const CSettingInt>(setting)->GetValue());
+    int maxPastDays = settings->GetInt(CSettings::SETTING_EPG_PAST_DAYSTODISPLAY);
+    CServiceBroker::GetPVRManager().Clients()->SetEPGTimeFrame(
+        maxPastDays, std::static_pointer_cast<const CSettingInt>(setting)->GetValue());
```

Comment:
Looking at this code sequence the question comes to mind, whether it would be better to have two separate API functions, one for future, one for past days. Why should we transfer the unchanged settings value to the addons?

Reply:
Yes, it is easy to have two calls. Let me make that change.

Reply:
Although then better to change the function names to `SetEPGMaxPastDays` and `SetEPGMaxFutureDays`.

Reply:
Ok, done @ksooo